tests/kafka-connect-jdbc/src/test/java/io/crate/qa/kafka/KafkaJdbcConnectorCrateDBIntegrationTest.java

```diff
@@ -0,0 +1,179 @@
+        assertThat(results.get(1), Is.is(m2));
+        assertThat(results.get(2), Is.is(m3));
+        assertThat(results.get(3), Is.is(m4));
+        assertThat(results.get(4), Is.is(m5));
+
+    }
+
+    ProducerRecord<Object, Object> toProducerRecord(Schema schema, Metrics metric) throws IOException {
+        return new ProducerRecord<>("metrics", toGenericRecord(schema, metric));
+    }
+
+    GenericRecord toGenericRecord(Schema schema, Metrics metric) throws IOException {
+        var datumWriter = new ReflectDatumWriter<>(schema);
+        var outputStream = new ByteArrayOutputStream();
+
+        var encoder = EncoderFactory.get().binaryEncoder(outputStream, null);
+        datumWriter.write(metric, encoder);
+        encoder.flush();
+
+        var datumReader = new GenericDatumReader<GenericRecord>(schema);
+        var decoder = DecoderFactory.get().binaryDecoder(outputStream.toByteArray(), null);
+        return datumReader.read(null, decoder);
+    }
+
+     List<Metrics> fetchResultsFromCrateDB() throws Exception {
+        try(var conn = DriverManager.getConnection(PSQL_DRIVER)) {
+            var stmt = conn.createStatement();
+            stmt.execute("refresh table metrics;");
+            try(var rs = stmt.executeQuery("select * from metrics order by id")) {
+                var results = new ArrayList<Metrics>();
+                while (rs.next()) {
+                    results.add(new Metrics(rs.getInt("id"), rs.getInt("x")));
+                }
+                return results;
+            }
+        }
+    }
+
+    void createMetricsTablesInCrateDB() throws Exception {
+        try(var conn = DriverManager.getConnection(PSQL_DRIVER)) {
+            var stmt = conn.createStatement();
+            stmt.execute("drop table if exists metrics;");
+            stmt.execute("create table metrics (id int primary key, x int);");
+        }
+    }
+
+    boolean setupKafkConnector() {
+        var configuration = Map.of(
+                "connector.class", "io.confluent.connect.jdbc.JdbcSinkConnector",
+                "topics", "metrics",
+                "connection.url", "jdbc:postgresql://cratedb:5432/doc?user=crate",
+                "tasks.max", 1,
+                "pk.mode", "record_value",
+                "pk.fields", "id",
+                "insert.mode", "upsert"
+        );
+
+        try {
+            var request = httpRequest()
+                    .PUT(HttpRequest.BodyPublishers.ofString(new ObjectMapper().writeValueAsString(configuration)))
+                    .build();
+            return HttpClient.newHttpClient().send(request, HttpResponse.BodyHandlers.ofString()).statusCode() == 200;
+        } catch (Exception ignored) {
+            return false;
+        }
+    }
+
+    void waitUntil(Supplier<Boolean> f) throws Exception {
+        for (int i = 1; i <= 10; i++) {
+            if (!f.get()) {
+                Thread.sleep(1000);
+            } else {
+                return;
+            }
+        }
```

Comment:
Maybe some exponentiall back-off could be used. But not sure if worth it - could help having faster tests on fast machines without overloading slow machines more.